internal/handlers/alumni_handler.go

```diff
@@ -0,0 +1,95 @@
+package handlers
+
+import (
+	"backend/internal/middleware"
+	"backend/internal/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type AlumniHandler struct {
+	db *gorm.DB
+}
+
+func NewAlumniHandler(db *gorm.DB) *AlumniHandler {
+	return &AlumniHandler{db: db}
+}
+
+func (h *AlumniHandler) Register(r *gin.RouterGroup) {
+	alumni := r.Group("/alumni")
+	{
+		// Public endpoints (require auth)
+		alumni.GET("", h.List)
+		alumni.GET("/:id", h.Get)
+
+		// Admin-only endpoints
+		alumni.Use(middleware.AdminRequired())
+		alumni.POST("", h.Create)
+		alumni.PUT("/:id", h.Update)
+		alumni.DELETE("/:id", h.Delete)
+
+	}
+}
+
+func (h *AlumniHandler) List(c *gin.Context) {
+	var alumni []models.Alumni
+	if err := h.db.Find(&alumni).Error; err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	c.JSON(http.StatusOK, alumni)
+}
+
+func (h *AlumniHandler) Get(c *gin.Context) {
+	id := c.Param("id")
+	var alumni models.Alumni
+	if err := h.db.First(&alumni, id).Error; err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Alumni not found"})
+		return
+	}
```

Comment:
Is this the only error this throws? It should return 404 if it's not found, and not InternalServerError. It should return InternalServerError if it's another error.